src/Symfony/Bundle/WebProfilerBundle/EventListener/TurboDriveCspListener.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebProfilerBundle\EventListener;
+
+use Symfony\Component\DependencyInjection\Attribute\When;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Show the Web Debug Toolbar (WDT) when navigating via Turbo Drive and when a strict Content-Security-Policy is set.
+ * This is done by reusing WDT's nonces.
+ * This service is registered only when TurboBundle is installed.
+ *
+ * @author David Petrásek <davidpetrasek@hotmail.cz>
+ */
+class TurboDriveCspListener implements EventSubscriberInterface
+{
+    public function onKernelResponse(ResponseEvent $event): void
+    {
+        $response = $event->getResponse();
+        $responseContent = $response->getContent();
+
+        if (!str_contains($responseContent, '<div id="sfwdt')) {
+            return;
+        }
+
+        $csp = $response->headers->get('Content-Security-Policy');
+        if (!$csp) {
+            return;
+        }
```

Comment:
```suggestion
        if (!$event->isMainRequest()) {
            return;
        }
        
        // do not capture redirects or modify XML HTTP Requests
        if ($request->isXmlHttpRequest()) {
            return;
        }
        
        if (!$response->headers->has('Content-Security-Policy')
            || $response->isRedirection()
            || ($response->headers->has('Content-Type') && !str_contains($response->headers->get('Content-Type') ?? '', 'html'))
            || 'html' !== $request->getRequestFormat()
            || false !== stripos($response->headers->get('Content-Disposition', ''), 'attachment;')
        ) {
            return;
        }

        if (!str_contains($responseContent, '<div id="sfwdt')) {
            return;
        }
        
```

Adapted quickly from the WebDebugToolbarListener.

The idea here is to exit early, only consider HTML pages and avoid calling getContent() on a binary response  file

Reply:
(missed a $response = $event->getResponse())